In [6]:
import pandas as pd
import numpy as np

In [7]:
df = pd.read_csv('./data/Aviation_Data.csv', parse_dates=['Event.Date', 'Publication.Date'], low_memory=False)

#Sam's addition, parsing dates

In [8]:
df.head()
# something looks off with event date because the years are not in range of data set

Event.Id Investigation.Type Accident.Number Event.Date  \
0  20001218X45444           Accident      SEA87LA080 1948-10-24   
1  20001218X45447           Accident      LAX94LA336 1962-07-19   
2  20061025X01555           Accident      NYC07LA005 1974-08-30   
3  20001218X45448           Accident      LAX96LA321 1977-06-19   
4  20041105X01764           Accident      CHI79FA064 1979-08-02   

          Location        Country   Latitude   Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2    Saltville, VA  United States  36.922223  -81.878056          NaN   
3       EUREKA, CA  United States        NaN         NaN          NaN   
4       Canton, OH  United States        NaN         NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaT  
1               UNK                Unknown  Probable Cause       1996-09-19  
2               IMC                 Cruise  Probable Cause       2007-02-26  
3               IMC                 Cruise  Probable Cause       2000-12-09  
4               VMC               Approach  Probable Cause       1980-04-16  

[5 rows x 31 columns]

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   Event.Id                88889 non-null  object        
 1   Investigation.Type      90348 non-null  object        
 2   Accident.Number         88889 non-null  object        
 3   Event.Date              88889 non-null  datetime64[ns]
 4   Location                88837 non-null  object        
 5   Country                 88663 non-null  object        
 6   Latitude                34382 non-null  object        
 7   Longitude               34373 non-null  object        
 8   Airport.Code            50249 non-null  object        
 9   Airport.Name            52790 non-null  object        
 10  Injury.Severity         87889 non-null  object        
 11  Aircraft.damage         85695 non-null  object        
 12  Aircraft.Category       32287 non-null  object

In [10]:
df.describe(include='all')

<ipython-input-10-174ba9bf1a5c>:1: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  df.describe(include='all')
<ipython-input-10-174ba9bf1a5c>:1: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  df.describe(include='all')


Event.Id Investigation.Type Accident.Number  \
count            88889              90348           88889   
unique           87951                 71           88863   
top     20001212X19172           Accident      ERA22LA119   
freq                 3              85015               2   
first              NaN                NaN             NaN   
last               NaN                NaN             NaN   
mean               NaN                NaN             NaN   
std                NaN                NaN             NaN   
min                NaN                NaN             NaN   
25%                NaN                NaN             NaN   
50%                NaN                NaN             NaN   
75%                NaN                NaN             NaN   
max                NaN                NaN             NaN   

                 Event.Date       Location        Country Latitude Longitude  \
count                 88889          88837          88663    34382     34373   
unique                14782          27758            219    25589     27154   
top     1982-05-16 00:00:00  ANCHORAGE, AK  United States  332739N  0112457W   
freq                     25            434          82248       19        24   
first   1948-10-24 00:00:00            NaN            NaN      NaN       NaN   
last    2022-12-29 00:00:00            NaN            NaN      NaN       NaN   
mean                    NaN            NaN            NaN      NaN       NaN   
std                     NaN            NaN            NaN      NaN       NaN   
min                     NaN            NaN            NaN      NaN       NaN   
25%                     NaN            NaN            NaN      NaN       NaN   
50%                     NaN            NaN            NaN      NaN       NaN   
75%                     NaN            NaN            NaN      NaN       NaN   
max                     NaN            NaN            NaN      NaN       NaN   

       Airport.Code Airport.Name  ... Purpose.of.flight Air.carrier  \
count         50249        52790  ...             82697       16648   
unique        10375        24871  ...                26       13590   
top            NONE      Private  ...          Personal       Pilot   
freq           1488          240  ...             49448         258   
first           NaN          NaN  ...               NaN         NaN   
last            NaN          NaN  ...               NaN         NaN   
mean            NaN          NaN  ...               NaN         NaN   
std             NaN          NaN  ...               NaN         NaN   
min             NaN          NaN  ...               NaN         NaN   
25%             NaN          NaN  ...               NaN         NaN   
50%             NaN          NaN  ...               NaN         NaN   
75%             NaN          NaN  ...               NaN         NaN   
max             NaN          NaN  ...               NaN         NaN   

       Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
count          77488.000000           76379.000000         76956.000000   
unique                  NaN                    NaN                  NaN   
top                     NaN                    NaN                  NaN   
freq                    NaN                    NaN                  NaN   
first                   NaN                    NaN                  NaN   
last                    NaN                    NaN                  NaN   
mean               0.647855               0.279881             0.357061   
std                5.485960               1.544084             2.235625   
min                0.000000               0.000000             0.000000   
25%                0.000000               0.000000             0.000000   
50%                0.000000               0.000000             0.000000   
75%                0.000000               0.000000             0.000000   
max              349.000000             161.000000           380.000

In [11]:
df['Investigation.Type'].value_counts()

Accident      85015
Incident       3874
25-09-2020      702
26-09-2020       60
02-02-2021       39
              ...  
05-04-2021        1
04-08-2021        1
11-02-2021        1
18-11-2020        1
05-01-2021        1
Name: Investigation.Type, Length: 71, dtype: int64

In [12]:
df['Purpose.of.flight'].value_counts()
# filter values for private vs commercial

Personal                     49448
Instructional                10601
Unknown                       6802
Aerial Application            4712
Business                      4018
Positioning                   1646
Other Work Use                1264
Ferry                          812
Aerial Observation             794
Public Aircraft                720
Executive/corporate            553
Flight Test                    405
Skydiving                      182
External Load                  123
Public Aircraft - Federal      105
Banner Tow                     101
Air Race show                   99
Public Aircraft - Local         74
Public Aircraft - State         64
Air Race/show                   59
Glider Tow                      53
Firefighting                    40
Air Drop                        11
ASHO                             6
PUBS                             4
PUBL                             1
Name: Purpose.of.flight, dtype: int64

In [13]:
df['Air.carrier'].value_counts()

Pilot                              258
American Airlines                   90
United Airlines                     89
Delta Air Lines                     53
SOUTHWEST AIRLINES CO               42
                                  ... 
Derek J. Craighead                   1
Dorothee                             1
Atlantic Southeast Airlines Inc      1
Aircraft Management Group            1
PARSONS CHARLIE C                    1
Name: Air.carrier, Length: 13590, dtype: int64

In [14]:
df['Location'].value_counts()
#filter data for country and state using .str.split()

ANCHORAGE, AK                      434
MIAMI, FL                          200
ALBUQUERQUE, NM                    196
HOUSTON, TX                        193
CHICAGO, IL                        184
                                  ... 
FINCASTLE, VA                        1
Beasley, TX                          1
ALMATY, Kazakhstan                   1
PUERTO BARRIOS, GUATEMALA            1
BUCKINGHAMSHIRE, UNITED KINGDOM      1
Name: Location, Length: 27758, dtype: int64

In [15]:
df['Total.Uninjured'].value_counts()

0.0      29879
1.0      25101
2.0      15988
3.0       4313
4.0       2662
         ...  
325.0        1
230.0        1
260.0        1
251.0        1
414.0        1
Name: Total.Uninjured, Length: 379, dtype: int64

# Sam's Addition

In [16]:
df.head(20)

Event.Id Investigation.Type Accident.Number Event.Date  \
0   20001218X45444           Accident      SEA87LA080 1948-10-24   
1   20001218X45447           Accident      LAX94LA336 1962-07-19   
2   20061025X01555           Accident      NYC07LA005 1974-08-30   
3   20001218X45448           Accident      LAX96LA321 1977-06-19   
4   20041105X01764           Accident      CHI79FA064 1979-08-02   
5   20170710X52551           Accident      NYC79AA106 1979-09-17   
6   20001218X45446           Accident      CHI81LA106 1981-08-01   
7   20020909X01562           Accident      SEA82DA022 1982-01-01   
8   20020909X01561           Accident      NYC82DA015 1982-01-01   
9   20020909X01560           Accident      MIA82DA029 1982-01-01   
10  20020909X01559           Accident      FTW82DA034 1982-01-01   
11  20020909X01558           Accident      ATL82DKJ10 1982-01-01   
12  20020917X02148           Accident      FTW82FRJ07 1982-01-02   
13  20020917X02134           Accident      FTW82FRA14 1982-01-02   
14  20020917X02119           Accident      FTW82FPJ10 1982-01-02   
15  20020917X02117           Accident      FTW82FPG08 1982-01-02   
16  20020917X01962           Accident      DEN82DTM08 1982-01-02   
17  20020917X01656           Accident      ANC82FAG14 1982-01-02   
18  20020917X02481           Accident      NYC82DA016 1982-01-02   
19  20020917X02339           Accident      MIA82DA028 1982-01-02   

            Location        Country   Latitude   Longitude Airport.Code  \
0    MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1     BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2      Saltville, VA  United States  36.922223  -81.878056          NaN   
3         EUREKA, CA  United States        NaN         NaN          NaN   
4         Canton, OH  United States        NaN         NaN          NaN   
5         BOSTON, MA  United States  42.445277  -70.758333          NaN   
6         COTTON, MN  United States        NaN         NaN          NaN   
7        PULLMAN, WA  United States        NaN         NaN          NaN   
8   EAST HANOVER, NJ  United States        NaN         NaN          N58   
9   JACKSONVILLE, FL  United States        NaN         NaN          JAX   
10         HOBBS, NM  United States        NaN         NaN          NaN   
11      TUSKEGEE, AL  United States        NaN         NaN          NaN   
12         HOMER, LA  United States        NaN         NaN          NaN   
13        HEARNE, TX  United States        NaN         NaN          T72   
14     CHICKASHA, OK  United States        NaN         NaN          NaN   
15   LITTLE ROCK, AR  United States        NaN         NaN          NaN   
16        MIDWAY, UT  United States        NaN         NaN          NaN   
17       SKWENTA, AK  United States        NaN         NaN          NaN   
18       GALETON, PA  United States        NaN         NaN          5G6   
19         MIAMI, FL  United States        NaN         NaN          NaN   

          Airport.Name  ... Purpose.of.flight Air.carrier  \
0                  NaN  ...          Personal         NaN   
1                  NaN  ...          Personal         NaN   
2                  NaN  ...          Personal         NaN   
3                  NaN  ...          Personal         NaN   
4                  NaN  ...          Personal         NaN   
5                  NaN  ...               NaN  Air Canada   
6                  NaN  ...          Personal         NaN   
7   BLACKBURN AG STRIP  ...          Personal         NaN   
8              HANOVER  ...          Business         NaN   
9    JACKSONVILLE INTL  ...          Personal         NaN   
10                 NaN  ...          Personal         NaN   
11            TUSKEGEE  ...          Personal         NaN   
12                 NaN  ...          Personal         NaN   
13    HEARNE MUNICIPAL  ...          Personal         NaN   
14                 NaN  ...          Personal         NaN   
15                 NaN  ...  

We want to string split so that we can lump by states and countries

In [17]:
df['Location_City'] = df['Location'].str.split(", ").str[0].str.upper()

In [18]:
df['Location_State_Country'] = df['Location'].str.split(", ").str[1].str.upper()

In [19]:
df['Location_State_Country'].value_counts().head()

CA    8857
TX    5913
FL    5825
AK    5672
AZ    2834
Name: Location_State_Country, dtype: int64

In [20]:
df.head()

Event.Id Investigation.Type Accident.Number Event.Date  \
0  20001218X45444           Accident      SEA87LA080 1948-10-24   
1  20001218X45447           Accident      LAX94LA336 1962-07-19   
2  20061025X01555           Accident      NYC07LA005 1974-08-30   
3  20001218X45448           Accident      LAX96LA321 1977-06-19   
4  20041105X01764           Accident      CHI79FA064 1979-08-02   

          Location        Country   Latitude   Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2    Saltville, VA  United States  36.922223  -81.878056          NaN   
3       EUREKA, CA  United States        NaN         NaN          NaN   
4       Canton, OH  United States        NaN         NaN          NaN   

  Airport.Name  ... Total.Fatal.Injuries Total.Serious.Injuries  \
0          NaN  ...                  2.0                    0.0   
1          NaN  ...                  4.0                    0.0   
2          NaN  ...                  3.0                    NaN   
3          NaN  ...                  2.0                    0.0   
4          NaN  ...                  1.0                    2.0   

  Total.Minor.Injuries Total.Uninjured Weather.Condition  \
0                  0.0             0.0               UNK   
1                  0.0             0.0               UNK   
2                  NaN             NaN               IMC   
3                  0.0             0.0               IMC   
4                  NaN             0.0               VMC   

  Broad.phase.of.flight   Report.Status  Publication.Date Location_City  \
0                Cruise  Probable Cause               NaT   MOOSE CREEK   
1               Unknown  Probable Cause        1996-09-19    BRIDGEPORT   
2                Cruise  Probable Cause        2007-02-26     SALTVILLE   
3                Cruise  Probable Cause        2000-12-09        EUREKA   
4              Approach  Probable Cause        1980-04-16        CANTON   

  Location_State_Country  
0                     ID  
1                     CA  
2                     VA  
3                     CA  
4                     OH  

[5 rows x 33 columns]

In [21]:
df.tail()

Event.Id Investigation.Type Accident.Number Event.Date  \
90343  20221227106491           Accident      ERA23LA093 2022-12-26   
90344  20221227106494           Accident      ERA23LA095 2022-12-26   
90345  20221227106497           Accident      WPR23LA075 2022-12-26   
90346  20221227106498           Accident      WPR23LA076 2022-12-26   
90347  20221230106513           Accident      ERA23LA097 2022-12-29   

            Location        Country Latitude Longitude Airport.Code  \
90343  Annapolis, MD  United States      NaN       NaN          NaN   
90344    Hampton, NH  United States      NaN       NaN          NaN   
90345     Payson, AZ  United States  341525N  1112021W          PAN   
90346     Morgan, UT  United States      NaN       NaN          NaN   
90347     Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Total.Fatal.Injuries Total.Serious.Injuries  \
90343          NaN  ...                  0.0                    1.0   
90344          NaN  ...                  0.0                    0.0   
90345       PAYSON  ...                  0.0                    0.0   
90346          NaN  ...                  0.0                    0.0   
90347          NaN  ...                  0.0                    1.0   

      Total.Minor.Injuries Total.Uninjured Weather.Condition  \
90343                  0.0             0.0               NaN   
90344                  0.0             0.0               NaN   
90345                  0.0             1.0               VMC   
90346                  0.0             0.0               NaN   
90347                  0.0             1.0               NaN   

      Broad.phase.of.flight Report.Status  Publication.Date Location_City  \
90343                   NaN           NaN        2022-12-29     ANNAPOLIS   
90344                   NaN           NaN               NaT       HAMPTON   
90345                   NaN           NaN        2022-12-27        PAYSON   
90346                   NaN           NaN               NaT        MORGAN   
90347                   NaN           NaN        2022-12-30        ATHENS   

      Location_State_Country  
90343                     MD  
90344                     NH  
90345                     AZ  
90346                     UT  
90347                     GA  

[5 rows x 33 columns]

After investigating the 'Investigation.Type' column, we have identified there should be only 2 categories ("Accident vs. Incident"), summing to a total of 88,889 rows vs. the total 90,348. We have looked at '25-09-2020', '26-09-2020', and '02-02-2021', all of which show rows consisting of null values. We need to filter out the dataset to show only the 2 categories.

In [26]:
df[df['Investigation.Type'] ==  '02-02-2021'].head()

Event.Id Investigation.Type Accident.Number Event.Date Location Country  \
75894      NaN         02-02-2021             NaN        NaT      NaN     NaN   
78228      NaN         02-02-2021             NaN        NaT      NaN     NaN   
78351      NaN         02-02-2021             NaN        NaT      NaN     NaN   
78499      NaN         02-02-2021             NaN        NaT      NaN     NaN   
78513      NaN         02-02-2021             NaN        NaT      NaN     NaN   

      Latitude Longitude Airport.Code Airport.Name  ... Total.Fatal.Injuries  \
75894      NaN       NaN          NaN          NaN  ...                  NaN   
78228      NaN       NaN          NaN          NaN  ...                  NaN   
78351      NaN       NaN          NaN          NaN  ...                  NaN   
78499      NaN       NaN          NaN          NaN  ...                  NaN   
78513      NaN       NaN          NaN          NaN  ...                  NaN   

      Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
75894                    NaN                  NaN             NaN   
78228                    NaN                  NaN             NaN   
78351                    NaN                  NaN             NaN   
78499                    NaN                  NaN             NaN   
78513                    NaN                  NaN             NaN   

      Weather.Condition Broad.phase.of.flight Report.Status  Publication.Date  \
75894               NaN                   NaN           NaN               NaT   
78228               NaN                   NaN           NaN               NaT   
78351               NaN                   NaN           NaN               NaT   
78499               NaN                   NaN           NaN               NaT   
78513               NaN                   NaN           NaN               NaT   

      Location_City Location_State_Country  
75894           NaN                    NaN  
78228           NaN                    NaN  
78351           NaN                    NaN  
78499           NaN                    NaN  
78513           NaN                    NaN  

[5 rows x 33 columns]

In [28]:
df = df[(df['Investigation.Type'] == 'Accident') | (df['Investigation.Type'] == 'Incident')]

In [29]:
#checking to see if df was filtered down to just Accidents and Incidents

df['Investigation.Type'].value_counts()

Accident    85015
Incident     3874
Name: Investigation.Type, dtype: int64

Next we want to investigate the date range earlier than 1962 because it seems like there are dates before this year.

In [30]:
df[(df['Event.Date'] < '1962-01-01')]

Event.Id Investigation.Type Accident.Number Event.Date  \
0  20001218X45444           Accident      SEA87LA080 1948-10-24   

          Location        Country Latitude Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States      NaN       NaN          NaN   

  Airport.Name  ... Total.Fatal.Injuries Total.Serious.Injuries  \
0          NaN  ...                  2.0                    0.0   

  Total.Minor.Injuries Total.Uninjured Weather.Condition  \
0                  0.0             0.0               UNK   

  Broad.phase.of.flight   Report.Status  Publication.Date Location_City  \
0                Cruise  Probable Cause               NaT   MOOSE CREEK   

  Location_State_Country  
0                     ID  

[1 rows x 33 columns]

There seems to be only one date before 1962. We should be able to throw out. However, the Aviation Accident Database provides this acknowledgement.

"Generally, a preliminary report is available online within a few days of an accident. Factual information is added when available, and when the investigation is completed, the preliminary report is replaced with a final description of the accident and its probable cause. Full narrative descriptions may not be available for dates before 1993, cases under revision, or where NTSB did not have primary investigative responsibility."

In [32]:
#We narrow the scope of our data to 1993 and after to start with data with full narrative descriptions

df = df[(df['Event.Date'] >= '1993-01-01')]

In [33]:
df.head()

Event.Id Investigation.Type Accident.Number Event.Date  \
32026  20001211X11729           Accident      MIA93FA041 1993-01-01   
32027  20001211X11711           Accident      LAX93LA087 1993-01-01   
32028  20001211X11645           Accident      ATL93LA045 1993-01-01   
32029  20001211X11732           Accident      MIA93LA042 1993-01-01   
32030  20001211X11647           Accident      ATL93LA048 1993-01-01   

                Location        Country Latitude Longitude Airport.Code  \
32026        LILLIAN, AL  United States      NaN       NaN          NaN   
32027         FOLSOM, CA  United States      NaN       NaN          NaN   
32028  FUQUAY-VARINA, NC  United States      NaN       NaN          5W5   
32029     TITUSVILLE, FL  United States      NaN       NaN          TIX   
32030       BESSEMER, AL  United States      NaN       NaN          2A3   

                 Airport.Name  ... Total.Fatal.Injuries  \
32026                     NaN  ...                  1.0   
32027                     NaN  ...                  0.0   
32028        TRIPLE W AIRPARK  ...                  0.0   
32029  SPACE CENTER EXECUTIVE  ...                  0.0   
32030                BESSEMER  ...                  0.0   

      Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
32026                    0.0                  1.0             0.0   
32027                    1.0                  0.0             0.0   
32028                    0.0                  1.0             0.0   
32029                    0.0                  1.0             1.0   
32030                    0.0                  0.0             2.0   

      Weather.Condition Broad.phase.of.flight   Report.Status  \
32026               VMC              Approach  Probable Cause   
32027               VMC           Maneuvering  Probable Cause   
32028               VMC              Approach  Probable Cause   
32029               VMC              Approach  Probable Cause   
32030               VMC               Takeoff  Probable Cause   

       Publication.Date  Location_City Location_State_Country  
32026        1993-11-15        LILLIAN                     AL  
32027        1993-10-13         FOLSOM                     CA  
32028        1993-10-25  FUQUAY-VARINA                     NC  
32029        1994-10-02     TITUSVILLE                     FL  
32030        1993-03-11       BESSEMER                     AL  

[5 rows x 33 columns]

In [34]:
#We take a look at where the null values are

df.isna().sum()

Event.Id                      0
Investigation.Type            0
Accident.Number               0
Event.Date                    0
Location                     41
Country                      73
Latitude                  22490
Longitude                 22499
Airport.Code              23443
Airport.Name              22694
Injury.Severity            1000
Aircraft.damage            2498
Aircraft.Category         28248
Registration.Number        1295
Make                         56
Model                        72
Amateur.Built               101
Number.of.Engines          5749
Engine.Type                7073
FAR.Description           28512
Schedule                  49496
Purpose.of.flight          6145
Air.carrier               41966
Total.Fatal.Injuries      11245
Total.Serious.Injuries    12309
Total.Minor.Injuries      11726
Total.Uninjured            5774
Weather.Condition          4489
Broad.phase.of.flight     26851
Report.Status              6381
Publication.Date           2784
Location

In [39]:
# We still have more cleaning up to do. Let's look at the country column
df['Country'].value_counts()

United States               50738
Brazil                        373
Canada                        350
United Kingdom                342
Mexico                        334
                            ...  
Yemen                           1
Mauritania                      1
Palau                           1
Turks and Caicos Islands        1
Niger                           1
Name: Country, Length: 213, dtype: int64

In [40]:
#Let's look at Weather Conditions
df['Weather.Condition'].value_counts()

VMC    48399
IMC     3335
UNK      378
Unk      262
Name: Weather.Condition, dtype: int64

There are 4 weather conditions (https://support.crewlounge.aero/support/solutions/articles/24000034167-difference-between-vfr-ifr-vmc-imc-and-actual-simulated-simulator-time)

VMC - Visual Meteorologic Condition: Visibility and clouds are clear and allow you to fly the aircraft by visual reference
IMC - Instrument Meteorologic Condition: The weather is restricting your visibility, you need aircraft flight instruments to fly safe

UNK and Unk means unknown. We should consolidate this into 1 grouping

In [41]:
df[df['Weather.Condition'] == "Unk"] = "UNK"

In [42]:
#Check the Weather Condition Categories
df['Weather.Condition'].value_counts()

VMC    48399
IMC     3335
UNK      640
Name: Weather.Condition, dtype: int64

Let's check for duplicate Accident Numbers

In [43]:
#Total Number of Accident Unfiltered is 56863
len(df['Accident.Number']

56863

In [48]:
#Check for Duplicates
df['Accident.Number'].value_counts().head()

UNK           262
ERA22FA318      2
DCA22WA167      2
DCA23WA071      2
CEN22LA149      2
Name: Accident.Number, dtype: int64

In [50]:
df[df['Accident.Number'] == 'UNK']

Event.Id Investigation.Type Accident.Number Event.Date Location Country  \
67173      UNK                UNK             UNK        UNK      UNK     UNK   
67763      UNK                UNK             UNK        UNK      UNK     UNK   
68347      UNK                UNK             UNK        UNK      UNK     UNK   
69021      UNK                UNK             UNK        UNK      UNK     UNK   
69509      UNK                UNK             UNK        UNK      UNK     UNK   
...        ...                ...             ...        ...      ...     ...   
90019      UNK                UNK             UNK        UNK      UNK     UNK   
90135      UNK                UNK             UNK        UNK      UNK     UNK   
90140      UNK                UNK             UNK        UNK      UNK     UNK   
90171      UNK                UNK             UNK        UNK      UNK     UNK   
90219      UNK                UNK             UNK        UNK      UNK     UNK   

      Latitude Longitude Airport.Code Airport.Name  ... Total.Fatal.Injuries  \
67173      UNK       UNK          UNK          UNK  ...                  UNK   
67763      UNK       UNK          UNK          UNK  ...                  UNK   
68347      UNK       UNK          UNK          UNK  ...                  UNK   
69021      UNK       UNK          UNK          UNK  ...                  UNK   
69509      UNK       UNK          UNK          UNK  ...                  UNK   
...        ...       ...          ...          ...  ...                  ...   
90019      UNK       UNK          UNK          UNK  ...                  UNK   
90135      UNK       UNK          UNK          UNK  ...                  UNK   
90140      UNK       UNK          UNK          UNK  ...                  UNK   
90171      UNK       UNK          UNK          UNK  ...                  UNK   
90219      UNK       UNK          UNK          UNK  ...                  UNK   

      Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
67173                    UNK                  UNK             UNK   
67763                    UNK                  UNK             UNK   
68347                    UNK                  UNK             UNK   
69021                    UNK                  UNK             UNK   
69509                    UNK                  UNK             UNK   
...                      ...                  ...             ...   
90019                    UNK                  UNK             UNK   
90135                    UNK                  UNK             UNK   
90140                    UNK                  UNK             UNK   
90171                    UNK                  UNK             UNK   
90219                    UNK                  UNK             UNK   

      Weather.Condition Broad.phase.of.flight Report.Status Publication.Date  \
67173               UNK                   UNK           UNK              UNK   
67763               UNK                   UNK           UNK              UNK   
68347               UNK                   UNK           UNK              UNK   
69021               UNK                   UNK           UNK              UNK   
69509               UNK                   UNK           UNK              UNK   
...                 ...                   ...           ...              ...   
90019               UNK                   UNK           UNK              UNK   
90135               UNK                   UNK           UNK              UNK   
90140               UNK                   UNK           UNK              UNK   
90171               UNK                   UNK           UNK              UNK   
90219               UNK                   UNK           UNK              UNK   

      Location_City Location_State_Country  
67173           UNK                    UNK  
67763           UNK                    UNK  
68347           UNK                    UNK  
69021           UNK                    UNK  
69509           UNK                    UNK  
...         

In [51]:
df[df['Accident.Number'] == 'ERA22FA318']

Event.Id Investigation.Type Accident.Number           Event.Date  \
89593  20220718105496           Accident      ERA22FA318  2022-07-17 00:00:00   
89594  20220718105496           Accident      ERA22FA318  2022-07-17 00:00:00   

            Location        Country Latitude Longitude Airport.Code  \
89593  Las Vegas, NV  United States  361239N  1151140W          VGT   
89594  Las Vegas, NV  United States  361239N  1151140W          VGT   

          Airport.Name  ... Total.Fatal.Injuries Total.Serious.Injuries  \
89593  NORTH LAS VEGAS  ...                    4                      0   
89594  NORTH LAS VEGAS  ...                    4                      0   

      Total.Minor.Injuries Total.Uninjured Weather.Condition  \
89593                    0               0               VMC   
89594                    0               0               VMC   

      Broad.phase.of.flight Report.Status     Publication.Date Location_City  \
89593                   NaN           NaN  2022-05-08 00:00:00     LAS VEGAS   
89594                   NaN           NaN  2022-05-08 00:00:00     LAS VEGAS   

      Location_State_Country  
89593                     NV  
89594                     NV  

[2 rows x 33 columns]

In [52]:
df[df['Accident.Number'] == 'DCA22WA167']

Event.Id Investigation.Type Accident.Number           Event.Date  \
88619  20220801105632           Incident      DCA22WA167  2022-01-22 00:00:00   
88620  20220801105632           Incident      DCA22WA167  2022-01-22 00:00:00   

       Location Country Latitude Longitude Airport.Code Airport.Name  ...  \
88619  Kigali,   Rwanda  001587S  0030822E          NaN          NaN  ...   
88620  Kigali,   Rwanda  001587S  0030822E          NaN          NaN  ...   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
88619                    0                      0                    0   
88620                    0                      0                    0   

      Total.Uninjured Weather.Condition Broad.phase.of.flight Report.Status  \
88619               0               NaN                   NaN           NaN   
88620               0               NaN                   NaN           NaN   

      Publication.Date Location_City Location_State_Country  
88619              NaT        KIGALI                         
88620              NaT        KIGALI                         

[2 rows x 33 columns]

Based on the investigation, we can remove the 'UNK' category in the Accident.Number, then we will need to drop duplicates

In [53]:
#First drop the UNK category

df = df[df['Accident.Number'] != 'UNK']

In [54]:
df['Accident.Number'].value_counts()

DCA22WA089    2
ERA22FA318    2
DCA22WA167    2
DCA23WA071    2
CEN22LA149    2
             ..
IAD03CA059    1
DCA13RA127    1
LAX08CA156    1
ERA09FA303    1
DCA20WA079    1
Name: Accident.Number, Length: 56575, dtype: int64

In [55]:
#Next drop duplicates
df = df.drop_duplicates(subset=['Accident.Number'])

In [56]:
df['Accident.Number'].value_counts()

ERA20CA190    1
ERA12LA058    1
LAX97LA090    1
NYC99FA161    1
ERA20LA322    1
             ..
DCA22LA201    1
WPR22FA004    1
ANC05CA146    1
NYC94LA017    1
DCA20WA079    1
Name: Accident.Number, Length: 56575, dtype: int64